### ISSUE TITLE
Failed to setup node following the quickstart guide

### ISSUE THREAD

**Issue by dhuang-esl** [ease-lab/vhive#473]
**Describe the bug**
On a fresh install ubuntu 20.04 image, the node setup script failed to run due to missing required dependencies.

**To Reproduce**
On a fresh install ubuntu 20.04 image, run the following commands to configure the node following the [quickstart guide:](https://github.com/ease-lab/vhive/blob/main/docs/quickstart_guide.md#1-setup-all-nodes)

```
sudo apt update
sudo apt install git

git clone --depth=1 https://github.com/ease-lab/vhive.git
cd vhive
mkdir -p /tmp/vhive-logs
./scripts/cloudlab/setup_node.sh > >(tee -a /tmp/vhive-logs/setup_node.stdout) 2> >(tee -a /tmp/vhive-logs/setup_node.stderr >&2)
```
Then the script `setup_node.sh` will drop several errors due to missing required dependencies. Please check Section [Logs](#logs) for detail information.

**Expected behavior**
Configure the node successfully

**Logs**
Terminal log:

```
ubuntu@ubuntu-KVM:~/Downloads/vhive$ ./scripts/cloudlab/setup_node.sh > >(tee -a /tmp/vhive-logs/setup_node.stdout) 2> >(tee -a /tmp/vhive-logs/setup_node.stderr >&2)
APT::Periodic::Update-Package-Lists "0";
APT::Periodic::Download-Upgradeable-Packages "0";
APT::Periodic::AutocleanInterval "0";
APT::Periodic::Unattended-Upgrade "0";
Created symlink /etc/systemd/system/apt-daily.service → /dev/null.
Created symlink /etc/systemd/system/apt-daily-upgrade.service → /dev/null.
/home/ubuntu/Downloads/vhive/scripts/setup_system.sh: line 28: curl: command not found
Warning: apt-key output should not be parsed (stdout is not a terminal)
gpg: no valid OpenPGP data found.
E: The repository 'https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_20.04  Release' does not have a Release file.
E: Unable to locate package skopeo
sysctl: setting key "net.ipv4.conf.all.promote_secondaries": Invalid argument
sudo: route: command not found
sudo: nft: command not found
sudo: nft: command not found
sudo: nft: command not found
sudo: nft: command not found
sudo: nft: command not found
sudo: nft: command not found
```

**Comment by ustiugov** [comment]
That's weird, the setup_system.sh script installs all the packages you mentioned. The script is used in the beginning of the quick start guide. I am not sure what breaks on your system

**Comment by dhuang-esl** [comment]
This is super weird because I just tested the [latest ubuntu 20.04.03 image downloaded from the ubuntu website](https://ubuntu.com/download/desktop) this morning, it still didn't work:
```
ubuntu@ubuntu-KVM:~/Downloads/vhive$ ./scripts/cloudlab/setup_node.sh > >(tee -a /tmp/vhive-logs/setup_node.stdout) 2> >(tee -a /tmp/vhive-logs/setup_node.stderr >&2)
APT::Periodic::Update-Package-Lists "0";
APT::Periodic::Download-Upgradeable-Packages "0";
APT::Periodic::AutocleanInterval "0";
APT::Periodic::Unattended-Upgrade "0";
Created symlink /etc/systemd/system/apt-daily.service → /dev/null.
Created symlink /etc/systemd/system/apt-daily-upgrade.service → /dev/null.
Warning: apt-key output should not be parsed (stdout is not a terminal)
OK
E: The repository 'https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_20.04  Release' does not have a Release file.
E: Unable to locate package skopeo
sysctl: setting key "net.ipv4.conf.all.promote_secondaries": Invalid argument
sudo: route: command not found
sudo: nft: command not found
sudo: nft: command not found
sudo: nft: command not found
sudo: nft: command not found
sudo: nft: command not found
sudo: nft: command not found
```
Here is the system info:
```
ubuntu@ubuntu-KVM:~/Downloads/vhive$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal
```

Do you have any idea?

**Comment by dhuang-esl** [comment]
Hi, I would like to provide more information about this bug. I found the workflow works very well on CloudLab with the provided CloudLab profile. However, it could not work for a fresh installed ubuntu image configured locally. I think maybe the image used in CloudLab is already pre-configured with some necessary dependencies before you can run the setup node script. Could you please also confirm this phenomenon?

**Comment by ustiugov** [comment]
@harshitgarg22 could you please take a look? Maybe download a fresh VM and try to reproduce it

**Comment by harshitgarg22** [comment]
Hi I can't reproduce this error. This is the output I get on an Azure VM, Cloudlab small-lan (different from the vhive one) profile for Ubuntu 20.04, Ubuntu VM using multipass.

```
ubuntu@foo:~/vhive$ ./scripts/cloudlab/setup_node.sh > >(tee -a /tmp/vhive-logs/setup_node.stdout) 2> >(tee -a /tmp/vhive-logs/setup_node.stderr >&2)
APT::Periodic::Update-Package-Lists "0";
APT::Periodic::Download-Upgradeable-Packages "0";
APT::Periodic::AutocleanInterval "0";
APT::Periodic::Unattended-Upgrade "0";
Created symlink /etc/systemd/system/apt-daily.service → /dev/null.
Created symlink /etc/systemd/system/apt-daily-upgrade.service → /dev/null.
Warning: apt-key output should not be parsed (stdout is not a terminal)
OK
sysctl: setting key "net.ipv4.conf.all.promote_secondaries": Invalid argument
containerd github.com/containerd/containerd v1.5.7 8686ededfc90076914c5238eb96c883ea093a8ba
Warning: apt-key output should not be parsed (stdout is not a terminal)
OK
🚧 Compile
net.bridge.bridge-nf-call-iptables  = 1
net.ipv4.ip_forward                 = 1
net.bridge.bridge-nf-call-ip6tables = 1
sysctl: setting key "net.ipv4.conf.all.promote_secondaries": Invalid argument
0 209715200 thin-pool /dev/loop4 /dev/loop3 128 32768 1 skip_block_zeroing
device-mapper: reload ioctl on fc-dev-thinpool  failed: No such device or address
Command failed.
```

```
ubuntu@foo:~/vhive$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.3 LTS
Release:        20.04
Codename:       focal
```

I remember having this error at some point but if I recall correctly it was fixed by running on a fresh install. 

Btw, the `Command failed` at the end is expected behaviour, and does not affect vhive program (as mentioned [here](https://github.com/ease-lab/vhive/blob/main/docs/quickstart_guide.md#1-setup-all-nodes)).

**Comment by dhuang-esl** [comment]
Hi @harshitgarg22! Thanks for trying it on Azure VM. It seems vHive can work with the ubuntu image provided by CloudLab and Azure. 

However, I encountered the bug while trying to configure vHive on a [ubuntu 20.04.3 image from the official Ubuntu website](https://ubuntu.com/download/desktop). Do you know if there is any difference between this image and the one provided by Cloudlab or Azure? Could you please also try to reproduce the error by using [the 20.04.3 image from the official Ubuntu website](https://ubuntu.com/download/desktop)?

**Comment by harshitgarg22** [comment]
I tried the ubuntu 20.04.03 and can confirm that the issue exists. 
I will try to find the solution for it.

**Comment by dhuang-esl** [comment]
Thanks to @harshitgarg22 and @ustiugov , this bug is fixed in #477
